packages/react-devtools-shared/src/devtools/views/Components/StackTraceView.js

```diff
@@ -63,13 +63,18 @@ export function CallSiteView({
   return (
     <div className={styles.CallSite}>
       {functionName || virtualFunctionName}
-      {' @ '}
-      <span
-        className={linkIsEnabled ? styles.Link : null}
-        onClick={viewSource}
-        title={url + ':' + line}>
-        {formatLocationForDisplay(url, line, column)}
-      </span>
+      {url !== '' && (
+        <>
+          {' @ '}
+          <span
+            className={linkIsEnabled ? styles.Link : null}
+            onClick={viewSource}
+            title={url + ':' + line}>
+            {formatLocationForDisplay(url, line, column)}
+          </span>
+        </>
+      )}
```

Comment:
How about displaying something like `Array.map @ native`? This is aligned with how Chrome displays V8's stacks.

I am not aware of other potential cases where url could be an empty string.

Reply:
Hmm, actually, I think we should extend `ReactStackTrace` to have boolean `isNative` instead. Because we already handle it here:
https://github.com/facebook/react/blob/d256ca013c5dd04529ba736ac7ca5f4d1c6d4186/packages/react-devtools-shared/src/backend/utils/parseStackTrace.js#L164-L167

And if its native, we should add `@ native` suffix, else if url is absent, then just don't display it.

Reply:
Sure, we can do that in a follow-up. We'd still want something for anonymous sources. Especially for the cases where we don't have access to structured stack traces.

Chrome isn't consistent with displaying the stacks anyway so it doesn't seem terribly important to be perfectly aligned. The `@` format is really just in tracing I think. Printing `.stack` doesn't use that formatting and use `at Array.map (<anonymous>)` instead.

Reply:
`isNative()` is also false for `Array.prototype.map` for me:
```
let error

const awaitedFlags = ['one'].map( (param) => {
    error = new Error()
    return 'a'
}
)

Error.prepareStackTrace = (error, structuredStackTrace) => {
    for (const frame of structuredStackTrace) {
        console.log(frame.toString(), frame.isNative())
    }
    return ''
}

error.stack
```